File: 898-transpose-matrix/transpose-matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
var transpose = function(matrix) {
    let n = matrix.length, m = matrix[0].length
    
    let nMat = [...new Array(m)].map(a => [...new Array(n)])


    for (let r = 0; r < n; r++) {
        for (let c = 0; c < m; c++) {
            nMat[c][r] = matrix[r][c]
        }
    }
    return nMat;
};

File: 898-transpose-matrix/transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows = len(matrix)
        if rows == 0: return matrix
        cols = len(matrix[0])

        grid = [[0 for _ in range(rows)] for _ in range(cols)]

        for r in range(rows):
            for c in range(cols):
                grid[c][r] = matrix[r][c]

        return grid